src/lazy/text/buffer.rs

```diff
@@ -934,4 +1007,45 @@ mod tests {
             mismatch_float(input);
         }
     }
+
+    #[test]
+    fn test_match_string() {
+        fn match_string(input: &str) {
+            MatchTest::new(input).expect_match(match_length(TextBufferView::match_string));
+        }
+        fn mismatch_string(input: &str) {
+            MatchTest::new(input).expect_mismatch(match_length(TextBufferView::match_string));
+        }
+
+        // These inputs have leading/trailing whitespace to make them more readable, but the string
+        // matcher doesn't accept whitespace. We'll trim each one before testing it.
+        let good_inputs = &[
+            r#"
+            "hello"
+            "#,
+            r#"
+            "😀😀😀"
+            "#,
+            r#"
+            "this has an escaped quote \" right in the middle"
+            "#,
+        ];
+        for input in good_inputs {
+            match_string(input.trim());
+        }
+
+        let bad_inputs = &[
+            // Missing an opening quote
+            r#"
+            hello"
+            "#,
+            // Missing a trailing quote
+            r#"
+            "hello
+            "#,
```

Comment:
The `mismatch` cases only look for a failure to match the entire string. We can add that test and it will fail (as it should) because it won't match the whole thing.